app/routes/claim/file-upload.js

```diff
@@ -20,52 +22,69 @@ module.exports = function (router) {
     csrfToken = generateCSRFToken(req)
 
     if (Object.prototype.hasOwnProperty.call(DocumentTypeEnum, req.params.documentType)) {
-      let claimId
-      if (req.query.document === 'VISIT_CONFIRMATION' || req.query.document === 'RECEIPT') {
-        claimId = req.params.claimId
-      } else {
-        claimId = null
-      }
-      DirectoryCheck(`${req.params.referenceId}-${req.query.eligibilityId}`, claimId, req.query.claimExpenseId, req.params.documentType)
       return res.render('claim/file-upload', {
         claimType: req.params.claimType,
         document: req.params.documentType,
         fileUploadGuidingText: DocumentTypeEnum,
         URL: req.url
       })
-    } else {
-      throw new Error('Not a valid document type')
     }
+
+    throw new Error('Not a valid document type')
   })
 
   router.post('/claim/file-upload/:referenceId/:claimId/:documentType', function (req, res, next) {
     authorisation.hasRoles(req, allowedRoles)
 
-    Upload(req, res, function (error) {
+    Upload(req, res, async function (error) {
       try {
         // If there was no file attached, we still need to check the CSRF token
         if (!req.file) {
```

Comment:
What happens when there is no file? 

Wonder if its possible to bail out early if there is no file rather than try to do some uploading and handling both cases lower down the file.

Reply:
It checks this in the callback. `Upload` is a Multer call and that puts it into `req.file` along with any errors etc. so would bail out in the second `if` in the `try`.